src/radio/radio-device-manager.ts

```diff
@@ -70,6 +70,35 @@ export class RadioDeviceManager implements smarthome.DeviceManager {
     );
   }
 
+  /**
+   * Fulfills a TcpRequestData.
+   * @param tcpRequestData  The TcpRequestData to source radio parameters from.
+   * @returns  A promise that resolves to the determined TcpResponseData.
+   */
+  private async processTcpRequestData(
+    tcpRequestData: smarthome.DataFlow.TcpRequestData
+  ): Promise<smarthome.DataFlow.TcpResponseData> {
+    const localAddress = this.deviceIdToAddress.get(tcpRequestData.deviceId);
+    const requestId = tcpRequestData.requestId;
+    const deviceId = tcpRequestData.deviceId;
+    const port = tcpRequestData.port;
+
+    if (tcpRequestData.operation === smarthome.Constants.TcpOperation.READ) {
+      const tcpResponse = await this.radioController.readTcpSocket(
+        localAddress!,
+        port
+      );
+      return new TcpResponseData(requestId, deviceId, tcpResponse);
+    } else {
```

Comment:
We assume write if not a read?

Reply:
That is the only other possible value for the type.  I added a comment to make it clear.